pkg/runtime/framework/plugins/jobset/builder.go

```diff
@@ -87,29 +93,58 @@ func (b *Builder) Initializer(trainJob *trainer.TrainJob) *Builder {
 }
 
 // Launcher updates JobSet values for the launcher Job.
-func (b *Builder) Launcher(info *runtime.Info, trainJob *trainer.TrainJob) *Builder {
+func (b *Builder) Launcher() *Builder {
```

Comment:
Why do we need Launcher builder in JobSet plugins if we just assign the "1" to the Job Replicas ?
Would it be simpler to just validate in webhook that `launcher` ReplicatedJob has `replicas: 1`?

Reply:
Yeah, that's right. We should add webhook validators for all reserved roles, Trainer, Initializer, and Launcher if replicatedJobs[*].replicas is `1`. However, I want to avoid conflicts with @akshaychitneni impls, and it sounds beyond the scope of this PR since we need to validate all reserved roles rather than just the Launcher.

So, I would like a follow-up.

Reply:
I opened an follow-up issue: https://github.com/kubeflow/trainer/issues/2502